File: hamersapp/src/main/java/nl/ecci/hamers/helpers/Fragments/DatePickerFragment.java
package nl.ecci.hamers.helpers.Fragments;

import android.app.DatePickerDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.widget.Button;
import android.widget.DatePicker;
import nl.ecci.hamers.R;

import java.util.Calendar;

public class DatePickerFragment extends DialogFragment implements DatePickerDialog.OnDateSetListener {

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        // Use the current date as the default date in the picker
        final Calendar c = Calendar.getInstance();
        int year = c.get(Calendar.YEAR);
        int month = c.get(Calendar.MONTH);
        int day = c.get(Calendar.DAY_OF_MONTH);

        // Create a new instance of DatePickerDialog and return it
        return new DatePickerDialog(getActivity(), this, year, month, day);
    }

    @Override
    public void onDateSet(DatePicker view, int year, int month, int day) {
        Button eventDateButton = (Button) getActivity().findViewById(R.id.event_date_button);
        Button eventEndDateButton = (Button) getActivity().findViewById(R.id.end_date_button);
        Button beerDateButton = (Button) getActivity().findViewById(R.id.pick_date_button);
        Button deadlineDateButton = (Button) getActivity().findViewById(R.id.deadline_date_button);

        String date = String.valueOf(day) + "-" + (month + 1) + "-" + year;

        if (getActivity().getSupportFragmentManager().findFragmentByTag("date") != null) {
            eventDateButton.setText(date);
        }

        if (getActivity().getSupportFragmentManager().findFragmentByTag("end_date") != null) {
            eventEndDateButton.setText(date);
        }

        if (getActivity().getSupportFragmentManager().findFragmentByTag("proefdatum") != null) {
            beerDateButton.setText(date);
        }
        if (getActivity().getSupportFragmentManager().findFragmentByTag("deadline_date") != null) {
            deadlineDateButton.setText(date);
        }
    }
}

File: hamersapp/src/main/java/nl/ecci/hamers/helpers/DatePickerFragment.kt
package nl.ecci.hamers.helpers

import android.app.DatePickerDialog
import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.widget.DatePicker
import kotlinx.android.synthetic.main.stub_new_event.*
import kotlinx.android.synthetic.main.stub_new_meeting.*
import kotlinx.android.synthetic.main.stub_new_review.*
import java.util.*

class DatePickerFragment : DialogFragment(), DatePickerDialog.OnDateSetListener {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        // Use the current date as the default date in the picker
        val c = Calendar.getInstance()
        val year = c.get(Calendar.YEAR)
        val month = c.get(Calendar.MONTH)
        val day = c.get(Calendar.DAY_OF_MONTH)

        // Create a new instance of DatePickerDialog and return it
        return DatePickerDialog(activity, this, year, month, day)
    }

    override fun onDateSet(view: DatePicker, year: Int, month: Int, day: Int) {
        val date = day.toString() + "-" + (month + 1) + "-" + year

        if (activity.supportFragmentManager.findFragmentByTag("date") != null) {
            event_date_button?.text = date
        }
        if (activity.supportFragmentManager.findFragmentByTag("end_date") != null) {
            event_end_date_button?.text = date
        }
        if (activity.supportFragmentManager.findFragmentByTag("proefdatum") != null) {
            pick_date_button?.text = date
        }
        if (activity.supportFragmentManager.findFragmentByTag("deadline_date") != null) {
            event_deadline_date_button?.text = date
        }
        if (activity.supportFragmentManager.findFragmentByTag("vergaderdatum") != null) {
            meeting_date_button?.text = date
        }
    }
}